test/k8sT/Services.go

```diff
@@ -830,14 +830,28 @@ var _ = Describe("K8sServicesTest", func() {
 				}
 			}
 
+			By("Removing %s pod so that another pod is chosen", dstPod)
+
 			// Delete the pod, and check that a new backend is chosen
-			nodes, err := kubectl.GetPodsNodes(helpers.DefaultNamespace, dstPod)
-			Expect(err).Should(BeNil(), "Cannot get node name of %s pod", dstPod)
 			kubectl.DeleteResource("pod", dstPod).ExpectSuccess("Unable to delete %s pod", dstPod)
-			// Wait until the pod has been removed from the cilium endpoints list.
-			// Otherwise, the requests below might fail as the non-existing endpoint
-			// will be chosen.
-			kubectl.WaitForCiliumEndpointDeleted(nodes[dstPod], helpers.DefaultNamespace, dstPod)
+
+			// Wait until the replacement pod has been provisioned and appeared
+			// in the ipcache of the second node.
+			//
+			// The first wait should give enough time for cilium-agents to remove
+			// the deleted pod from the BPF LB maps, so that the next request won't
+			// choose the deleted pod.
+			//
+			// The second wait is needed to make sure that an IPCache entry of the
+			// new pod appears on the k8s1 node. Otherwise, if the new pod runs
+			// on k8s2 and a request below selects it, the request will be dropped
+			// in the vxlan mode (the tailcall IPV4_NODEPORT_NAT body won't pass
+			// the request to the encap routines, and instead it will be dropped
+			// due to failing fib_lookup).
+			waitPodsDs()
+			if fromOutside && vxlan {
+				time.Sleep(5 * time.Second)
```

Comment:
Would it make sense to create a new helper to wait for the specific ipcache entry, so that we are sure that it has been added before going on? Maybe overkill, I'm not strongly opposed to the 5s wait.

Reply:
It's a bit an overkill, but on the other hand we are risking to introduce flakes with sleep. Going to add it in this PR.